### PR TITLE
VM Engine Performance Improvements

### PR DESCRIPTION
**Pull request by brianlenz**
cc: @ixje 

**What current issue(s) does this address, or what feature is it adding?**

Excessive CPU usage on RPC nodes: #352 

Note that this only helps avoid some unnecessary calls, but it doesn't solve the CPU load issue entirely.

**How did you solve this problem?**

Removed unnecessary calls to `write_log` and `ToName`.

**How did you make sure your solution works?**

Profiled performance before and after over a similar time period. The number of invocations reduced from 130k to 9.

**Are there any special changes in the code that we should be aware of?**

No.

**Please check the following, if applicable:**

- [ ] Did you add any tests?
- [X] Did you run `make lint`?
- [X] Did you run `make test`?
- [X] Are you making a PR to a feature branch or development rather than master?
- [ ] Did you add an entry to `CHANGELOG.rst`? (if not, please do)

### CODE REVIEW

**Comment by localhuman** [comment]
I cant remember if debug vm is on by default.  If so it should probably be off.

**Comment by brianlenz** [comment]
Pretty sure it is off by default which is why this change is needed and had an impact.

**Comment by brianlenz** [comment]
Confirmed it's disabled by default, which is why this fix is needed to remove the excessive and unnecessary `ToName` invocations. You have to run `config vm-log on` to enable it.

**Comment by localhuman** [comment]
Another simple change is to remove this line from `api_server.py` https://github.com/CityOfZion/neo-python/blob/master/neo/bin/api_server.py#L196

**Comment by brianlenz** [comment]
Why would we want to get rid of the background logging thread?

**Comment by metachris** [comment]
Yeah I wouldn't remove that, it also has neglectible impact on performance, but is very useful to see block counts in the logs.

File: neo/VM/ExecutionEngine.py [hunk 1]
```diff
@@ -37,14 +37,17 @@ class ExecutionEngine():
     # file descriptor
     log_file = None
 
+    def is_write_log(self):
```
Comment (localhuman) [inline neo/VM/ExecutionEngine.py]: I think one way to further speed things up determine on `__init__` a property called `_is_write_log` which is just the value of `settings.log_vm_instructions`.  

Then in line 912 you can just do `if self._is_write_log: ...` and in the `write_log` method you can check if the log file exists and if its not closed before actually writing.

Comment (brianlenz) [reply neo/VM/ExecutionEngine.py]: Sure thing; done!

**Pull request merged**